### PR TITLE
client-go: attach request source with retry info for coprocessor  (#46706)

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

Cherry pick #46706 to standalone branch.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: store/copr/coprocessor.go [hunk 1]
```diff
@@ -272,6 +272,10 @@ type copTask struct {
 
 	// timeout value for one kv readonly request
 	tikvClientReadTimeout uint64
+	// timeout value for one kv readonly reqeust
+	tidbKvReadTimeout uint64
+	// firstReadType is used to indicate the type of first read when retrying.
+	firstReadType string
```
Comment (cfzjywxk) [inline store/copr/coprocessor.go:278]: When is the `firstReadType` initialized to a non-empty value for coprocessor tasks?

**Review by zyguan — APPROVED** [review]

File: store/copr/coprocessor.go (hunk 1 above)
Comment (zyguan) [reply store/copr/coprocessor.go:276]: [here](https://github.com/pingcap/tidb/pull/47003/files#diff-23eb1043649369d5956b123b173b799ded8c6b7cbf4519bb32cb3caa01928d75R1091), and `req.ReadType` will be initialized by replica selector.

**Review by ekexium — APPROVED** [review]

**Pull request merged**